src/vs/workbench/contrib/notebook/browser/view/cellParts/codeCell.ts

```diff
@@ -148,10 +148,12 @@ export class CodeCell extends Disposable {
 		this._register({ dispose() { cts.dispose(true); } });
 		raceCancellation(this.viewCell.resolveTextModel(), cts.token).then(model => {
 			if (this._isDisposed) {
+				model?.dispose();
 				return;
 			}
 
 			if (model) {
+				this._register(model);
```

Comment:
This seems legit, but I'll leave it to someone else to make the call

Reply:
@roblourens I'm not sure if it's a good idea to dispose the model since `BaseCellViewModel` has a model ref  `_textModelRef`. Disposing the model here would leave us an obsolete `_textModelRef` and `textModel`.